vulnerabilities/importers/safety_db.py

```diff
@@ -103,29 +101,40 @@ def updated_advisories(self) -> Set[Advisory]:
         advisories = []
 
         for package_name in self._api_response:
+            if package_name == "$meta":
+                # This is the first entry in the data feed. It contains metadata of the feed.
+                # Skip it.
+                continue
+
+            try:
+                validate_schema(self._api_response[package_name])
+
+            except Exception as e:
+                logger.error(e)
+                continue
+
             all_package_versions = self.versions.get(package_name)
-            if len(all_package_versions) == 0:
+            if not len(all_package_versions):
                 # PyPi does not have data about this package, we skip these
                 continue
 
             for advisory in self._api_response[package_name]:
-
                 impacted_purls, resolved_purls = categorize_versions(
                     package_name, all_package_versions, advisory["specs"]
                 )
 
-                cve_ids = advisory.get("cve") or [""]
-
-                # meaning if cve_ids is not [''] but either ['CVE-123'] or ['CVE-123, CVE-124']
-                if len(cve_ids[0]):
-                    cve_ids = [s.strip() for s in cve_ids.split(",")]
+                if advisory["cve"]:
+                    # Check on advisory["cve"] instead of using `get` because it can have null value
+                    cve_ids = re.findall(r"CVE-\d+-\d+", advisory["cve"])
```

Comment:
nit: FWIW, I tend to prefer having this as external functions defined this way:
```
import re

get_cve_ids = re.compile(r"CVE-\d+-\d+", re.IGNORE_CASE).findall
```
Then you can use get_cve_ids and test it too

Reply:
I think this belongs to a helpers module, from where we can reuse this

Reply:
Doing this in other PR